src/libfsm/gen.c

```diff
@@ -562,8 +568,8 @@ sfs_step_edges(struct gen_ctx *ctx, struct gen_stack_frame *sf)
 		sf->u.step_edges.initialized = true;
 	}
 
-	if (ctx->buf_used + ctx->sed[sf->s_id] >= ctx->max_length) {
-		LOG(2, "PRUNING due to max length: used:%zu + sed[%d]:%u >= max_length:%zu\n",
+	if (ctx->buf_used + ctx->sed[sf->s_id] > ctx->max_length) {
+		LOG(2, "PRUNING due to max length: used:%zu + sed[%d]:%u > max_length:%zu\n",
```

Comment:
@silentbicycle requesting your review for this please, I was unsure about this.

Reply:
I think `>` is correct here -- this means "if the buffer used so far plus what we'd need for the shortest distance to an end state exceeds the available buffer". `re -G 5 '^abcde'` should produce "abcde" as a valid 5-character match (which is rejected with `>=`), not counting the trailing `\0`, but reject `re -G 5 '^abcdef'`.